bridge/client/app/_components/ktb-trigger-sequence/ktb-trigger-sequence.component.ts

```diff
@@ -0,0 +1,363 @@
+    }
+
+    if (this.sequenceType === TRIGGER_SEQUENCE.EVALUATION) {
+      this.triggerEvaluation();
+    }
+
+    if (this.sequenceType === TRIGGER_SEQUENCE.CUSTOM) {
+      this.triggerCustomSequence();
+    }
+  }
+
+  private getImageString(image: string, tag: string): string {
+    return image.replace(/\s/g, '') + ':' + tag.replace(/\s/g, '');
+  }
+
+  private parseTimeframe(timeframe: Timeframe): string {
+    let timeframeString = '';
+    timeframeString += timeframe.hours ? timeframe.hours + 'h' : '';
+    timeframeString += timeframe.minutes ? timeframe.minutes + 'm' : '';
+    timeframeString += timeframe.seconds ? timeframe.seconds + 's' : '';
+    timeframeString += timeframe.millis ? timeframe.millis + 'ms' : '';
+    timeframeString += timeframe.micros ? timeframe.micros + 'us' : '';
+
+    return timeframeString;
+  }
+
+  private parseLabels(labels: string): { [key: string]: string } {
+    const labelObj: { [key: string]: string } = {};
+    const lbls = labels.replace(/\s/g, '').split(',');
+    for (const label of lbls) {
+      const parts = label.split('=');
+      if (parts[1]) {
+        labelObj[parts[0]] = parts[1];
+      }
+    }
+
+    return labelObj;
+  }
+
+  private triggerDelivery(): void {
+    const data: TriggerSequenceData = {
+      project: this.projectName || '',
+      stage: this.selectedStage || '',
+      service: this.selectedService || '',
+    };
+
+    if (this.deliveryFormData.labels && this.deliveryFormData.labels.trim() !== '') {
+      data.labels = this.parseLabels(this.deliveryFormData.labels);
+    }
+
+    if (this.deliveryFormData.values) {
+      const valuesObj = JSON.parse(this.deliveryFormData.values);
+      data.configurationChange = {
+        values: {
+          ...valuesObj,
+          image: this.getImageString(this.deliveryFormData.image || '', this.deliveryFormData.tag || ''),
+        },
+      };
+    } else {
+      data.configurationChange = {
+        values: {
+          image: this.getImageString(this.deliveryFormData.image || '', this.deliveryFormData.tag || ''),
+        },
+      };
+    }
+
+    this.dataService.triggerDelivery(data).subscribe(
+      (res) => {
+        this.handleResponse(res);
+      },
+      (err) => {
+        this.isLoading = false;
+        console.log(err);
+      }
+    );
+  }
+
+  private triggerEvaluation(): void {
+    const data: TriggerEvaluationData = {
+      project: this.projectName || '',
+      stage: this.selectedStage || '',
+      service: this.selectedService || '',
+      evaluation: {},
+    };
+    if (this.evaluationFormData.labels && this.evaluationFormData.labels.trim() !== '') {
+      data.evaluation.labels = this.parseLabels(this.evaluationFormData.labels);
+    }
+
+    if (
+      this.evaluationFormData.evaluationType === TRIGGER_EVALUATION_TIME.TIMEFRAME &&
+      this.evaluationFormData.timeframe
+    ) {
+      data.evaluation.timeframe = this.parseTimeframe(this.evaluationFormData.timeframe);
```

Comment:
I would move the `&& this.evaluationFormData.timeframe` inside this if and check the next if with else if instead. Because if the type is set to `TIMEFRAME`, it can't be `START_END`